### PR TITLE
Remove unnecessary condition else check

### PR DESCRIPTION
**Pull request by Akanksha-kedia**
## Description
Eliminate unneeded else condition check

## Motivation and Context
Streamlining and eliminating superfluous if-else conditions.

## Impact
No impact, just removal of extraneous elements.

## Test Plan
 Execute unit tests in presto-main and verify castToTimestamp and castToTime scenarios.

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== RELEASE NOTES ==

Remove unnecessary condition else check

### CODE REVIEW

**Comment by Akanksha-kedia** [comment]
2023-11-21T00:20:32.223-0600    WARN    TestNG-test=Surefire test-1     org.eclipse.jetty.util.thread.QueuedThreadPool  QueuedThreadPool[http-worker]@5fbaf2de{STOPPING,2<=20<=200,i=0,q=12}[org.eclipse.jetty.util.thread.TryExecutor$$Lambda$280/1814466511@18707d06] Couldn't stop Thread[http-worker-59,5,main]
[INFO] Tests run: 158, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 373.087 s - in com.facebook.presto.hive.TestHiveDistributedJoinQueries
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 158, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------

**Comment by Akanksha-kedia** [comment]
2023-11-21T00:58:06.843-0600    WARN    TestNG-test=Surefire test-1     com.facebook.presto.testng.services.LogTestDurationListener     Tests from com.facebook.presto.verifier.framework.TestDataVerification took 1.88m
[INFO] Tests run: 180, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 236.171 s - in TestSuite
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 180, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  04:38 min

**Comment by Akanksha-kedia** [comment]
@tdcmeehan  please review

File: presto-main/src/main/java/com/facebook/presto/type/TimestampWithTimeZoneOperators.java [hunk 1]
```diff
@@ -146,9 +146,8 @@ public static long castToTime(SqlFunctionProperties properties, @SqlType(Standar
         if (properties.isLegacyTimestamp()) {
             return modulo24Hour(unpackChronology(value), unpackMillisUtc(value));
         }
-        else {
-            return modulo24Hour(castToTimestamp(properties, value));
-        }
+        //Redundant else
```
[the remark this answers is not in the document]
Comment (aaneja) [reply presto-main/src/main/java/com/facebook/presto/type/TimestampWithTimeZoneOperators.java:149]: You can simplify this further to - 
```
return properties.isLegacyTimestamp() ? modulo24Hour(unpackChronology(value), unpackMillisUtc(value)) : modulo24Hour(castToTimestamp(properties, value));
```

**Review by aaneja — APPROVED** [review]

**Comment by Akanksha-kedia** [comment]
@tdcmeehan please helpt o review

**Review by tdcmeehan — APPROVED** [review]

**Pull request merged**